cluster/consensus_module_proxy_compat.go

```diff
@@ -0,0 +1,182 @@
+package cluster
+
+import (
+	"github.com/lirm/aeron-go/aeron/atomic"
+	"github.com/lirm/aeron-go/aeron/logbuffer"
+	"github.com/lirm/aeron-go/cluster/codecs"
+)
+
+// -----------------------------------------------------------------------------
+// Java compat implementation and using TryClaim for better performance
+// https://github.com/real-logic/aeron/blob/release/1.46.x/aeron-cluster/src/main/java/io/aeron/cluster/service/ConsensusModuleProxy.java
+// -----------------------------------------------------------------------------
+
+// https://github.com/real-logic/aeron/blob/release/1.46.x/aeron-cluster/src/main/java/io/aeron/cluster/service/ConsensusModuleProxy.java#L146
+func (proxy *consensusModuleProxy) ack(
+	logPosition int64,
+	timestamp int64,
+	ackID int64,
+	relevantID int64,
+	serviceID int32,
+) (bool, error) {
+	length := int32(new(codecs.MessageHeader).EncodedLength() + int64(new(codecs.ServiceAck).SbeBlockLength()))
```

Comment:
nit: seems a bit inefficient to allocate a struct to get a constant. In [java code it is a static field](https://github.com/real-logic/aeron/blob/release/1.46.x/aeron-cluster/src/main/java/io/aeron/cluster/service/ConsensusModuleProxy.java#L149C38-L150C73)